plugins/logs/plugin.go

```diff
@@ -575,6 +580,12 @@ func (p *Plugin) Log(ctx context.Context, decision *server.Info) error {
 		}
 	}
 
+	if p.config.Service != "" {
+		p.mtx.Lock()
+		defer p.mtx.Unlock()
```

Comment:
nit: I think we should unlock before calling into the plugin. Otherwise, we're effectively serializing plugin execution.